src/lib.rs

```diff
@@ -143,33 +191,21 @@ struct DebugInfo<'object, Endian>
 {
     debug_line: gimli::DebugLine<'object, Endian>,
     units: Vec<Unit<'object, Endian>>,
+    with_demangling: bool,
```

Comment:
The more I see this, the more I want demangling to stay outside the internals of the lookup code.  For example, it would be entirely sufficient for `Mapping::locate` to do the demangling. And this would avoid the need to propagate down the `with_demangling` bool.